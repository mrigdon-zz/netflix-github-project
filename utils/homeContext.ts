import React from "react";
import { ColumnId } from "../components/OrgReposTable";

export interface HomeParams {
  search: string;
  ascending: boolean;
  column: ColumnId;
}

export const HomeContext = React.createContext<{
  params: HomeParams;
  setParams: (params: HomeParams) => void;
}>({
  params: { search: "", ascending: false, column: "stars" },
  setParams(_: HomeParams) {},
});
