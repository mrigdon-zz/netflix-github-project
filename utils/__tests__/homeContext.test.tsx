import { useContext, useState } from "react";
import { act, create } from "react-test-renderer";
import { HomeContext, HomeParams } from "../homeContext";

function Consumer() {
  const { params, setParams } = useContext(HomeContext);
  return (
    <div>
      <button
        onClick={() =>
          setParams({ search: "foo", ascending: true, column: "updated" })
        }
      >
        change stuff
      </button>
      {JSON.stringify(params, null, 2)}
    </div>
  );
}

function Provider({ initialParams }: { initialParams: HomeParams }) {
  const [params, setParams] = useState<HomeParams>(initialParams);
  return (
    <HomeContext.Provider value={{ params, setParams }}>
      <Consumer />
    </HomeContext.Provider>
  );
}

describe("HomeContext", () => {
  it("provides context to the consumer", () => {
    const renderer = create(
      <Provider
        initialParams={{
          search: "",
          ascending: false,
          column: "stars",
        }}
      />
    );
    expect(renderer).toMatchSnapshot("before changing");

    act(() => renderer.root.findByType("button").props.onClick());

    expect(renderer).toMatchSnapshot("after changing");
  });
});
