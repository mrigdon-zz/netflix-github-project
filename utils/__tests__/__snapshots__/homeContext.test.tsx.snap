// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContext provides context to the consumer: after changing 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    change stuff
  </button>
  {
  "search": "foo",
  "ascending": true,
  "column": "updated"
}
</div>
`;

exports[`HomeContext provides context to the consumer: before changing 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    change stuff
  </button>
  {
  "search": "",
  "ascending": false,
  "column": "stars"
}
</div>
`;
